fix(module:tree): read initial state of nzNodes

rajniszp opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Node state (`isExpanded`, `isSelected`, `isChecked`, `isHalfChecked`, `isMatched`) is ignored when updating nodes/loading children

Issue Number: N/A

## What is the new behavior?
Node state is correctly set and thus represented on the UI

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No


## Other information

rajniszp committed:
fix(module:tree): read initial state of nzNodes

rajniszp committed:
fix(module:tree): read initial state of nzNodes
read state recursively (including all descendant nodes)

Nicoss54 reviewed (changes requested):
Could you had some test please

rajniszp committed:
fix(module:tree): read initial state of nzNodes
getting half-checked nodes from checked nodes, setting matchedNodeList not needed (no "matched" in NzTreeNodeOptions); added tests

rajniszp commented:
Added tests analogous to existing ones
(failed CodeFactor check highlights lines that I did not change)
